toolchain/sem_ir/ids.h

```diff
@@ -112,6 +112,55 @@ struct BoolValue : public IndexBase, public Printable<BoolValue> {
 constexpr BoolValue BoolValue::False = BoolValue(0);
 constexpr BoolValue BoolValue::True = BoolValue(1);
 
+// The ID of a name. A name is either a string or a special name such as
+// `self`, or eventually `Self` or `base`.
+struct NameId : public IndexBase, public Printable<NameId> {
+  // An explicitly invalid ID.
+  static const NameId Invalid;
+  // The name of `self`.
+  static const NameId SelfValue;
+  // The name of `Self`.
+  static const NameId SelfType;
+  // The name of the return slot in a function.
+  static const NameId ReturnSlot;
+
+  // Returns the NameId corresponding to a particular IdentifierId.
+  static auto ForIdentifier(IdentifierId id) -> NameId {
+    static_assert(NameId::InvalidIndex == IdentifierId::InvalidIndex);
+    CARBON_CHECK(id.index >= 0 || id.index == InvalidIndex)
+        << "Unexpected identifier ID";
+    return NameId(id.index);
+  }
+
+  // Returns the IdentifierId corresponding to this NameId, or an invalid
+  // IdentifierId if this is a special name.
+  auto AsIdentifierId() -> IdentifierId {
+    return index >= 0 ? IdentifierId(index) : IdentifierId::Invalid;
+  }
+
+  using IndexBase::IndexBase;
+  auto Print(llvm::raw_ostream& out) const -> void {
+    out << "name";
+    if (*this == Invalid) {
+      out << "Invalid";
```

Comment:
Why is `Invalid` printed this way versus the default `<invalid>` that `IndexBase::Print` would give?

Reply:
Fixed.